__std__/flake.nix

```diff
@@ -0,0 +1,248 @@
+        # Not all cells have the same organelles.
+        # All organelles belong in a cell.
+        #
+        # In this repository we have five organelles, listed below with their type:
+        #   devshells :: devshells
+        #     Development shells available via nix develop
+        #   packages :: installables
+        #     Packages available via nix build
+        #   devshellProfiles :: functions
+        #     Building blocks for devshells, not exposed by the flake
+        #   scripts :: functions
+        #     Bash scripts simplifying or automating a variety of tasks
+        #     Generally these are available as commands inside the development shell
+        #     These are very repository specific, and are not exposed by the flake
+        #   library :: functions
+        #     Functions and values shared across the current cell
+        #     These are very repository specific, and are not exposed by the flake
+        #
+        # std provides a TUI to interact with the organelles.
+        # Available interactions are determined by the organelle's type.
+        # Because this repository does not use the TUI, the type is actually irrelevant.
+        organelles = [
+          (inputs.std.devshells "devshells")
+          (inputs.std.installables "packages")
+          (inputs.std.functions "devshellProfiles")
+          (inputs.std.functions "scripts")
+          (inputs.std.functions "library")
+        ];
+      }
+
+      # The growOn function will then accept an arbitrary number of attrs.
+      # This is where we translate cells and organelles into a standard
+      # nix flake outputs attrs.
+      #
+      # This is where we also decide which cells and which organelles will
+      # make it into the flake. To exclude stuff from the flake, we simply
+      # do not "harvest" it.
+      #
+      # The attrs will be recursively merged in the order in which they appear.
+      {
+        # Here we say that we want the devshells organelle of the doc cell
+        # (which contains a number of shell-able derivations) to be exposed
+        # by the flake and accessible via nix develop.
+        devShells = inputs.std.harvest inputs.self [ "doc" "devshells" ];
+
+        # Here we say that we want the packages organelle of the doc cell
+        # (which contains a number of buildable derivations) to be exposed
+        # by the flake and accessible via nix build (or nix run).
+        packages = inputs.std.harvest inputs.self [ "doc" "packages" ];
+      }
+      {
+        # The devshells inside the haskell cells will be added to the ones
+        # already harvested from the doc shell. Same for packages.
+        devShells = inputs.std.harvest inputs.self [ "haskell" "devshells" ];
+        packages = inputs.std.harvest inputs.self [ "haskell" "packages" ];
+      }
+      {
+        packages = inputs.std.harvest inputs.self [ "toolchain" "packages" ];
+      };
+
+  # # # # # THE STANDARD FORMAT OF NIX FILES
```

Comment:
I really fancied the idea of fitting everything in flake.nix
However this isn't too pleasing to the eye... And it will probably grow larger.
I'll add this as a TODO, to move the docs into a README, once refactoring is completed.